modules/virt-accessing-vmi-yaml-ssh.adoc

```diff
@@ -0,0 +1,147 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vmi-yaml-ssh_{context}"]
+= Accessing a virtual machine instance via SSH with YAML configurations
+
+You can enable a SSH connection to a VMI without the need to run the `virtctl expose` command. The YAML file for the VMI should include a label with a value that is identical to the value for the `selector` in the service YAML file. Then, the service forwards the SSH traffic connections to the VMI.
+
+The following examples show the configurations for the VMI's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* You must be in the same project as the VMI.
+* The VMI you want to access must be connected to the default pod network by using the `masquerade` binding method.
+
+.Procedure
+. In the YAML file for the VMI, add the label and a value for exposing the service for SSH connections. Add the port number and enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+kind: VirtualMachineInstance
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-ssh
+  name: vmi-ssh <1>
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {} <2>
+        name: testmasquerade <3>
+        ports:
+        - name: ssh
+          port: 22 <4>
+          protocol: TCP
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+...
+----
+<1> The name for the VMI that is used to expose the details for the SSH service. This label is used by the service to identify the virtual machine instances that are enabled for SSH traffic connections.
+<2> The interface type is `masquerade`.
+<3> The name of this interface is `testmasquerade`.
+<4> The port number used for SSH connections.
+
+. Create the VMI:
++
+[source,terminal]
+----
+oc create -f <path_for_the_VMI_YAML_file>
+----
++
+. Create the service:
++
+[source,terminal]
+----
+oc create -f <path_for_the_service_YAML_file>
+----
++
+. Verify that the VMI is running:
++
+[source,terminal]
+----
+oc get vmi
+----
++
+.Example output
+[source,terminal]
+----
+NAME    AGE     PHASE       IP              NODENAME
+vmi-ssh 6s      Running     10.244.196.152  node01
+----
+
+. Create the YAML file for the SSH service:
++
+[source,terminal]
+----
+$ oc create -f vmi-ssh.yaml
+----
+
++
+.Example output
```

Comment:
I think the part where we explain about  the yaml of the service should come before we create the service(line 76). What do you think?

Reply:
Agree.

Reply:
Agree, that makes sense.

Reply:
so I think that the yaml definition below should come before the `oc create -f <path for the service yaml file` command.

The yaml file is not the output of `oc create -f`, rather the yaml should be properly defined before we use `oc create -f`.